src/Microsoft.AspNet.Routing/RouteBase.cs

```diff
@@ -237,11 +238,12 @@ public virtual VirtualPathData GetVirtualPath(VirtualPathContext context)
             }
         }
 
-        private void EnsureBinder()
+        private void EnsureBinder(HttpContext context)
         {
             if (_binder == null)
             {
-                _binder = new TemplateBinder(ParsedTemplate, Defaults);
+                var urlEncoder = context.RequestServices.GetRequiredService<UrlEncoder>();
+                _binder = new TemplateBinder(ParsedTemplate, urlEncoder, Defaults);
```

Comment:
@ryanbrandenburg had to add it here - https://github.com/aspnet/Mvc/commit/70f9431e7fd500583a84821cbcae4a97f64f78b2. Not sure why it would be missing in that case. I'll go ahead and do a TryAdd since we require it. It'd make it more reliable.